pytest_pyodide/decorator.py

```diff
@@ -383,6 +384,7 @@ def __init__(
         packages: Collection[str] = (),
         pytest_assert_rewrites: bool = True,
         *,
+        skip_if_not_all_built: bool = "CI" in os.environ,
```

Comment:
I think setting the default to `"CI" in os.environ` might be confusing to users, as they will silently get different behavior in local and in CI. How about 1) setting the default to `False`, and make it configurable in the [CLI option](https://github.com/pyodide/pytest-pyodide/blob/1670627699308e875639f767f8470a44349587a5/pytest_pyodide/hook.py#L126)

Reply:
Makes sense. Also, it seems that what we really want is to skip if it's in one of the earlier test jobs (e.g., `test_packages_no_numpy_dependents`) but fail if it's in the final job.

Reply:
> if it's in one of the earlier test jobs (e.g., test_packages_no_numpy_dependents) but fail if it's in the final job.

Yeah, makes sense.